src/vcpkg/archives.cpp

```diff
@@ -67,6 +67,26 @@ namespace vcpkg::Archives
                                code_and_output.output);
             recursion_limiter_sevenzip_old = false;
         }
+        else if (ext == ".msi")
+        {
+            // msiexec is a WIN32/GUI application, not a console application and so needs special attention to wait
+            // until it finishes (wrap in cmd /c).
+            const auto code_and_output = cmd_execute_and_capture_output(
+                Command{"cmd"}
+                    .string_arg("/c")
+                    .string_arg("msiexec")
+                    .string_arg("/a")
+                    .path_arg(archive)
+                    .string_arg("/qn")
+                    // msiexec requires quotes to be after "TARGETDIR=":
+                    //      TARGETDIR="C:\full\path\to\dest"
+                    .raw_arg(Strings::concat("TARGETDIR=", Command{to_path_partial}.extract())));
```

Comment:
This has been tested with spaces as well:

```
[DEBUG] CreateProcessW(cmd /c msiexec /a "C:\src\vcpkg\down loads\7z1900-x64.msi" /qn TARGETDIR="C:\src\vcpkg\down loads\tools\7zip-19.00-windows.partial.11052")       
[DEBUG] cmd_execute_and_stream_data() returned 0 after   735266 us
C:\src\vcpkg\down loads\tools\7zip-19.00-windows\Files\7-Zip\7z.exe
```

Reply:
Why not add an actual e2e test for that?